cli/tests/pcluster/configure/test_pcluster_configure.py

```diff
@@ -21,60 +21,90 @@ def _mock_input(mocker, input_in_order):
     mocker.patch(UTILS + "input", side_effect=input_in_order)
 
 
-def _mock_aws_region(mocker):
-    regions = [
-        "eu-north-1",
-        "ap-south-1",
-        "eu-west-3",
-        "eu-west-2",
-        "eu-west-1",
-        "ap-northeast-2",
-        "ap-northeast-1",
-        "sa-east-1",
-        "ca-central-1",
-        "ap-southeast-1",
-        "ap-southeast-2",
-        "eu-central-1",
-        "us-east-1",
-        "us-east-2",
-        "us-west-1",
-        "us-west-2",
-    ]
-    mocker.patch(EASYCONFIG + "get_regions", return_value=regions)
-
-
-def _mock_list_keys(mocker):
+def _mock_aws_region(mocker, partition="commercial"):
+    regions = {
+        "commercial": [
+            "eu-north-1",
+            "ap-south-1",
+            "eu-west-3",
+            "eu-west-2",
+            "eu-west-1",
+            "ap-northeast-2",
+            "ap-northeast-1",
+            "sa-east-1",
+            "ca-central-1",
+            "ap-southeast-1",
+            "ap-southeast-2",
+            "eu-central-1",
+            "us-east-1",
+            "us-east-2",
+            "us-west-1",
+            "us-west-2",
+        ],
+        "china": ["cn-north-1", "cn-northwest-1"],
+    }
+    mocker.patch(EASYCONFIG + "get_regions", return_value=regions.get(partition))
+
+
+def _mock_list_keys(mocker, partition="commercial"):
     # If changed look for test_prompt_a_list
-    keys = ["key1", "key2", "key3", "key4", "key5", "key6"]
-    mocker.patch(EASYCONFIG + "_get_keys", return_value=keys)
+    keys = {
+        "commercial": ["key1", "key2", "key3", "key4", "key5", "key6"],
+        "china": ["some_key1", "some_key2", "some_key3"],
+    }
+    mocker.patch(EASYCONFIG + "_get_keys", return_value=keys.get(partition))
 
 
-def _mock_list_vpcs_and_subnets(mocker, empty_region=False):
+def _mock_list_vpcs_and_subnets(mocker, empty_region=False, partition="commercial"):
     # If changed look for test_prompt_a_list_of_tuple
     if empty_region:
         mocked_response = {"vpc_list": [], "vpc_to_subnets": {}}
     else:
-        mocked_response = {
-            "vpc_list": [
-                ("vpc-12345678", "ParallelClusterVPC-20190625135738", "2 subnets inside"),
-                ("vpc-23456789", "ParallelClusterVPC-20190624105051", "0 subnets inside"),
-                ("vpc-34567891", "default", "3 subnets inside"),
-                ("vpc-45678912", "ParallelClusterVPC-20190626095403", "1 subnets inside"),
-            ],
-            "vpc_subnets": {
-                "vpc-12345678": [
-                    ("subnet-12345678", "ParallelClusterPublicSubnet", "Subnet size: 256"),
-                    ("subnet-23456789", "ParallelClusterPrivateSubnet", "Subnet size: 4096"),
+        response_dict = {
+            "commercial": {
```

Comment:
you could add another "fake" or "empty" key to return the empty region response, and avoid to have another flag:

```python
_mock_list_vpcs_and_subnets(mocker, partition="commercial"):
 vpcs = {
    "commercial": {...},
    "china": {...},
    "fake|empty": {"vpc_list": [], "vpc_to_subnets": {}}

}
```

Reply:
We are passing in partition when we call `MockHandler(mocker, partition="china")` which calls all the`_mock...` functions with `partition`, so we will need a third variable for the special case of empty vpc and subnets anyway.